### PR TITLE
fix(google): Fix for the missing CredentialsRepository issue

### PR DESCRIPTION
**Pull request by kirangodishala**
This fix addresses the issue - https://github.com/spinnaker/spinnaker/issues/6733

### CODE REVIEW

File: clouddriver-google/src/main/java/com/netflix/spinnaker/clouddriver/google/config/GoogleCredentialsConfiguration.java [hunk 1]
```diff
@@ -111,4 +116,13 @@ public void synchronize() {
       }
     };
   }
+
+  @Bean
+  @ConditionalOnMissingBean(
+      value = GoogleNamedAccountCredentials.class,
+      parameterizedContainer = CredentialsRepository.class)
+  public CredentialsRepository<GoogleNamedAccountCredentials> googleCredentialsRepository(
```
Comment (dbyron-sf) [inline clouddriver-google/src/main/java/com/netflix/spinnaker/clouddriver/google/config/GoogleCredentialsConfiguration.java:124]: I realize it's potentially painful, but I gotta ask...what's involved in adding a test for this?

**Comment by dbyron-sf** [comment]
@kirangodishala https://github.com/spinnaker/clouddriver/pull/5747 is the PR that introduced the bug, right?

**Comment by kirangodishala** [comment]
Yes @dbyron-sf. And this PR is the fix for that.

**Comment by kirangodishala** [comment]
@dbyron-sf - As this is issue occurs only after all the configuration and beans are loaded, this can't be verified through a unit test. Will work on integration tests  under [buildtool](https://github.com/spinnaker/buildtool) repo.  Could you please merge this PR so that the clouddriver would be usable.

**Comment by dbyron-sf** [comment]
> @dbyron-sf - As this issue occurs only after all the configuration and beans are loaded, this can't be verified through a unit test. Will work on integration tests under [buildtool](https://github.com/spinnaker/buildtool) repo. Could you please merge this PR so that the clouddriver would be usable.

I do think it's possible to test this using ApplicationContextRunner, like [this](https://github.com/spinnaker/clouddriver/blob/25bed1fa85a9f75dcdc4faa453fb33baf2d7b8aa/clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesSearchProviderTest.java) or [this](https://github.com/spinnaker/clouddriver/blob/25bed1fa85a9f75dcdc4faa453fb33baf2d7b8aa/clouddriver-core/src/test/java/com/netflix/spinnaker/clouddriver/cache/CatsSearchProviderTest.java).

**Comment by dbyron-sf** [comment]
As for timing, I'm hoping for a fix for [one more issue](https://github.com/spinnaker/spinnaker/issues/6734) before releasing 1.29.1.  I won't let the absence of tests hold that up.

**Comment by dbyron-sf** [comment]
Thanks much @kirangodishala !

**Review by dbyron-sf — APPROVED** [review]

**Comment by dbyron-sf** [comment]
@mergifyio backport release-1.29.x

**Pull request merged**